pkg/rtc/uptrackmanager.go

```diff
@@ -298,29 +308,64 @@ func (u *UpTrackManager) getPublishedTrack(trackID livekit.TrackID) types.MediaT
 	return u.publishedTracks[trackID]
 }
 
-func (u *UpTrackManager) parseSubscriptionPermissions(subscriptionPermission *livekit.SubscriptionPermission) {
+func (u *UpTrackManager) parseSubscriptionPermissions(
+	subscriptionPermission *livekit.SubscriptionPermission,
+	resolver func(participantID livekit.ParticipantID) types.LocalParticipant,
+) error {
 	// every update overrides the existing
 
 	// all_participants takes precedence
 	if subscriptionPermission.AllParticipants {
 		// everything is allowed, nothing else to do
 		u.subscriberPermissions = nil
-		return
+		return nil
 	}
 
 	// per participant permissions
-	u.subscriberPermissions = make(map[livekit.ParticipantID]*livekit.TrackPermission)
+	u.subscriberPermissions = make(map[livekit.ParticipantIdentity]*livekit.TrackPermission)
 	for _, trackPerms := range subscriptionPermission.TrackPermissions {
-		u.subscriberPermissions[livekit.ParticipantID(trackPerms.ParticipantSid)] = trackPerms
+		subscriberIdentity := livekit.ParticipantIdentity(trackPerms.ParticipantIdentity)
+		if subscriberIdentity == "" {
+			if trackPerms.ParticipantSid == "" {
+				u.subscriberPermissions = nil
+				return ErrSubscriptionPermissionNeedsId
+			}
+
+			var sub types.LocalParticipant
+			if resolver != nil {
+				sub = resolver(livekit.ParticipantID(trackPerms.ParticipantSid))
+			}
+			if sub == nil {
+				u.params.Logger.Warnw("could not find subscriber for permissions update", nil, "subscriberID", trackPerms.ParticipantSid)
+				continue
+			}
+
+			subscriberIdentity = sub.Identity()
+		} else {
+			if trackPerms.ParticipantSid != "" {
```

Comment:
I think these types of errors are tricky because the client would not be made aware of it.

should we log the error but still to honor identity?

Reply:
sure can make the change.